lib/vandelay/models/base.rb

```diff
@@ -9,7 +9,7 @@ def initialize(**data)
         data.each do |prop, val|
           self.instance_variable_set "@#{prop}", val
           self.class.class_eval do
-            define_method("#{prop}") { val }
+            attr_accessor prop
```

Comment:
@coryjamescrook nice discussion during today's interview.  yes, this is indeed a better approach than using eval.